assets/js/baseline-grid.js

```diff
@@ -0,0 +1,108 @@
+/* global Prince */
+
+// Lint with JS Standard
+
+function ebAlignToGrid (querySelectorString) {
+  // Get all elements we want aligned with baseline grid.
+  // Note that unlike InDesign we do not force elements
+  // to the next baseline unit. Rather, we add margin-bottom
+  // so that they end aligned. Then any following elements
+  // will also be aligned with the baseline grid.
+  // This is mainly because Prince collapses margin-top
+  // on elements that appear at the top of a page,
+  // and we can't add margin-top to those, nor distinguish them
+  // from other elements in order to use a different algorithm.
+  // We use margin, not padding, because in testing it has been
+  // more reliable (e.g. padding on a table doesn't add space).
+
+  let elementsToAlign = document.querySelectorAll('.content h1, .content h2, .content h3, .content h4, .content h5, .content math[display="block"], .content p, .content ol, .content ul, .content table, .content .figure, .content .video, .content .box, .content .align-to-baseline')
+
+  const elementsToLeave = document.querySelectorAll('.content li > p, .content .figure *')
+
+  if (querySelectorString) {
+    elementsToAlign = document.querySelectorAll(querySelectorString)
+  }
+
+  if (elementsToAlign) {
+    console.log('Aligning elements to baseline grid...')
+
+    // Get the baseline grid. We assume it is the line-height
+    // applied to the body element. Need to convert CSS strings to numbers.
+    const gridValuePt = window.getComputedStyle(document.body).lineHeight
+    const gridValue = parseFloat(gridValuePt.replace('pt', ''))
```

Comment:
Do we need to add an error handling step here, in case someone has not defined line-height-default in units of pt?

Reply:
@LouiseSteward Good question. I've done some testing, and confirmed that, in Prince, `window.getComputedStyle(element).lineHeight` always returns a value in `pt`, much like browsers return `px`, even if the CSS uses a different unit. 

Still, I'll generalise to strip any unit because that's better code, and I'll add a code comment.